### ISSUE TITLE
Allow variables in `(:include ...)`

### ISSUE THREAD

**Issue by samoht** [ocaml/dune#3416]
It would be nice if this would work:

```
... (:include %{lib:<dir>:file}) ...
```

That would avoid to have to do that little dance every time:

```
(rule (copy_file %{lib:<dir>:file} file)

... (:include file) ...
```

**Comment by jeremiedimino** [comment]
We wanted to get rid of the `(:include ...)` syntax entirely though...

**Comment by samoht** [comment]
> We wanted to get rid of the (:include ...) syntax entirely though...

To replace it by what? :-)

**Comment by jeremiedimino** [comment]
By `%{read:<file>}`, `%{read-lines:<file>}`, etc... This two mechanisms feel very similar

**Comment by samoht** [comment]
ha yes that would work too! 

But it doesn't seem possible to write `%{read:lib:<pkg>:file}` either right now.

**Comment by jeremiedimino** [comment]
Indeed. TBH, I'm just a bit worried that we are extending the syntax in all directions in ad-hoc ways. i.e., we are slowly implementing the famous bug-ridden lisp interpreter...

**Comment by rgrinberg** [comment]
bug-ridden lisp interpreter or not, I think Jeremie implemented this in 3.0
